### PR TITLE
fix(index): enable hnsw.iterative_scan for project-filtered ArchGuard search

### LINKED ISSUE #64: Enable hnsw.iterative_scan for project-filtered ArchGuard search
## Problem

`PgStore.Search` (`internal/index/pgvector.go`) filters a single shared HNSW index (`archguard_adrs_embedding_idx`) by `project_name` after the ANN graph walk, not before. Benchmark data gathered for #44 (see #63, `BenchmarkPgStoreSearch_ProjectFiltering`) shows this causes real, scale-dependent recall loss:

| Scale | Baseline recall | `iterative_scan` recall |
|---|---|---|
| 1 project | 100% | 100% |
| 10 projects, 25 ADRs | 88% | 100% |
| 10 projects, 100 ADRs | 72.7% | 86% |
| 50 projects, 25 ADRs | 28% | 98% |
| 50 projects, 100 ADRs | 23.3% | 80% |

At 50 projects sharing the table, the current default configuration returns fewer than one relevant ADR per query on average (`results_per_query` 0.82-0.84 out of 3 requested) — meaning real architectural drift can go unflagged with no error or warning, purely because other tenants' ADRs share the same index.

pgvector's `hnsw.iterative_scan = 'relaxed_order'` GUC (0.8.0+, resolved version confirmed as 0.8.5 via #57) recovers most of this recall at a bounded latency cost (roughly break-even at low scale, up to ~2.3x p50 at the worst scale point tested, still sub-4ms p50 in absolute terms).

## Acceptance Criteria

- Enable `hnsw.iterative_scan = 'relaxed_order'` for connections `PgStore` uses to run `Search` — either unconditionally in `PgStore`'s connection setup, or behind a new `vector_store` config knob (e.g. `vector_store.iterative_scan`), consistent with this repo's existing `*bool`-with-nil-default pattern for optional-but-defaulting-on settings (see `vector_store.reindex_enabled`/`reindex_concurrently` in `internal/config`).
- Guard against pgvector versions that don't support the GUC (pre-0.8.0): setting an unrecognized GUC name doesn't error in Postgres (confirmed during #63's benchmark development — this is exactly what made the original `SET`-based version-detection approach unsound), so a version check via `pg_extension.extversion` is required before setting it, not just a bare `SET`/`ALTER ROLE`.
- Document the new behavior/config knob in `CLAUDE.md`'s `internal/index` section and `archguard.yaml`'s reference config, alongside the existing `reindex_*` settings.
- Add integration test coverage confirming `hnsw.iterative_scan` is actually applied to `PgStore`'s connections (e.g. via `EXPLAIN`/`SHOW`, similar to `assertUsesHNSWIndex` in `pgvector_bench_test.go`).

## Related

- #44 — investigation and benchmark that produced this recommendation.
- #63 — benchmark implementation (`BenchmarkPgStoreSearch_ProjectFiltering`), reusable for validating this fix's recall improvement directly.
- #57 — pgvector version pinning (relevant to the version-guard requirement above).

### PR DESCRIPTION
**Pull request by Tgenz1213**
## Summary

Closes #64. `PgStore.Search` filters one shared HNSW index by `project_name` *after* the ANN graph walk, which a benchmark from #63 measured as losing real recall as tenant count grows (down to 23-28% recall at 50 tenants) — meaning real architectural drift could go unflagged with no error, purely because other tenants' ADRs share the index.

This enables pgvector's `hnsw.iterative_scan = 'relaxed_order'` GUC on every real `PgStore` connection, gated by:
- a new `vector_store.iterative_scan` config knob (`*bool`, nil-default-enabled, same pattern as `vector_store.reindex_enabled`)
- a runtime `pg_extension.extversion` check (0.8.0+ required) — a bare `SET`/`ALTER ROLE` can't detect support, since Postgres doesn't error on an unrecognized GUC name

- `internal/index/pgvector.go`: `ReindexOptions` renamed to `HNSWOptions` (+ new `IterativeScan` field) since it now covers more than reindex tuning; version probe runs once in `NewPgStore` on the existing extension-creation connection; the GUC is applied per-connection via `AfterConnect`; a `Pool()` accessor was added so tests can inspect real pooled-connection state (an `AfterConnect`-applied GUC only shows up on connections that actually came from that pool).
- New integration tests prove the GUC is actually applied on a real pooled connection (`SHOW hnsw.iterative_scan`), not just that config compiles.
- New ADR (`docs/arch/0005-hnsw-iterative-scan-for-project-filtered-search.md`), CLAUDE.md updates, and an `archguard.yaml` example block (which also documents the pre-existing `reindex_*` knobs there for the first time — they'd never had an example in the reference config either).

## Deviations from plan

- **`AfterConnect` failure mode**: the plan originally specified propagating a `SET hnsw.iterative_scan` failure the same way `RegisterTypes`'s error propagates (hard-failing the connection). A later review round flagged this as inconsistent with every other failure path in the same function (version-probe error, unsupported version both degrade gracefully) — a restricted-GUC managed Postgres could turn a best-effort recall enhancement into a hard connectivity failure. Changed to warn-and-continue instead.
- **Benchmark baseline fix**: `pgvector_bench_test.go`'s `"baseline"` sub-benchmark (added in #63) constructed `PgStore` with `HNSWOptions{}`, which now defaults to iterative-scan *enabled* — silently contaminating the "baseline" arm with the mitigation it was supposed to measure without. Fixed to explicitly disable it for that arm.
- A few smaller fixes came out of a second review pass (external `/code-review high`): a version-string-parsing edge-case inconsistency (unreachable in practice, but worth closing), a stale CLAUDE.md sentence about which test files require Docker, de-duplicating the pgvector-version-probe SQL between production and benchmark code (mirroring this repo's existing `SearchQuery`-export precedent), and a version-aware skip on the new integration test so it doesn't fail against a hypothetical future pre-0.8.0 pgvector image.
- Two findings from that same external review were investigated and rejected as out of scope: a claim that `PgStore.Close()` was newly added and unused (`Close()` predates this PR, from #63) and a claim this PR's diff opens an unclosed second connection pool in `internal/cli/cli.go` (that file isn't touched anywhere in this branch — the double-`NewVectorStore` call pattern there is pre-existing and unrelated to #64).
- Issue #57 (pin the `pgvector/pgvector:pg16` Docker tag) is explicitly out of scope — the runtime `pg_extension.extversion` check works regardless of whether the tag is pinned, so this PR doesn't depend on it.

## Test plan

- [x] `go build -o archguard ./cmd/archguard` succeeds
- [x] `go test -v -race -cover ./...` passes
- [x] `golangci-lint run --timeout=5m` clean
- [x] With Docker running locally: all `internal/index` integration/bench tests pass against real pgvector 0.8.5, including the new `SHOW hnsw.iterative_scan` tests (enabled → `relaxed_order`, explicitly disabled → `off`)
- [x] `BenchmarkPgStoreSearch_ProjectFiltering` re-run post-fix: baseline vs. iterative_scan arms now genuinely diverge (e.g. 50 projects/100 ADRs: 16.0% vs 78.7% recall), consistent with the recall numbers in #64
- [ ] CI: this repo's `internal/index` package spins up its own Postgres+pgvector via testcontainers-go (no Postgres service container in `.github/workflows/ci.yml`), so the Docker-gated tests above should run for real in CI too — worth confirming the CI run actually exercises them rather than skipping silently

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/index/pgvector.go [hunk 1]
```diff
@@ -45,18 +79,46 @@ func NewPgStore(connStr string, projectName string, concurrency int, reindex Rei
...
+	switch {
+	case versionErr != nil:
+		if iterativeScanWanted {
+			fmt.Printf("Warning: failed to check pgvector version for hnsw.iterative_scan support (%v); leaving it disabled for this connection.\n", versionErr)
+		}
+	case iterativeScanWanted && IterativeScanSupportedVersion(pgvectorVersion):
+		applyIterativeScan = true
+	case iterativeScanWanted:
+		fmt.Printf("Warning: pgvector %s does not support hnsw.iterative_scan (requires 0.8.0+); project-filtered search recall may be degraded at scale. See docs/arch/0005-hnsw-iterative-scan-for-project-filtered-search.md.\n", pgvectorVersion)
+	}
```
[the remark this answers is not in the document]
Comment (Tgenz1213) [reply internal/index/pgvector.go:105]: Fixed in 050850f — reworded to "leaving it disabled for all connections from this store" to reflect that this fires once in NewPgStore before the pool exists and gates every connection the store ever opens, not a single one.

While in there, also fixed a related wording issue on the per-connection SET-failure warning a few lines below (flagged separately as a suppressed comment in the review summary): it said the connection would fall back to "exact HNSW search," which is wrong — a failed SET just means that connection runs standard non-iterative HNSW search (still approximate, same as before this PR), not exact/brute-force search. Reworded to "standard (non-iterative) HNSW search."
